src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,289 @@
-# An implementation would go here!
+import subprocess
+from os import X_OK, access
+from zipfile import BadZipFile, ZipFile
+
+from requests import exceptions as requests_exceptions
+
+from briefcase.commands import (
+    BuildCommand,
+    CreateCommand,
+    PackageCommand,
+    PublishCommand,
+    RunCommand,
+    UpdateCommand,
+)
+from briefcase.config import BaseConfig
+from briefcase.exceptions import BriefcaseCommandError, NetworkFailure
+from briefcase.integrations import adb
+
+
+class ApkMixin:
+    output_format = "apk"
+    platform = "android"
+    # Storing this here so instances can override it for testing.
+    adb = adb
+
+    def binary_path(self, app):
+        return (
+            self.platform_path
+            / app.formal_name
+            / "app"
+            / "build"
+            / "outputs"
+            / "apk"
+            / "debug"
+            / "app-debug.apk"
+        )
+
+    def distribution_path(self, app):
+        return self.binary_path(app)
+
+    @property
+    def sdk_path(self):
+        return self.dot_briefcase_path / "tools" / "android_sdk"
+
+    @property
+    def sdk_url(self):
+        """The Android SDK URL appropriate to the current operating system."""
+        # The URLs described by the pattern below have existed since
+        # approximately 2017, and the code they download has a built-in
+        # updater. I hope they will work for many years.
+        return "https://dl.google.com/android/repository/" + (
+            "sdk-tools-{os}-4333796.zip".format(os=self.host_os.lower())
+        )
+
+    def verify_python_version(self):
+        if self.python_version_tag != "3.7":
+            raise BriefcaseCommandError(
+                """\
+Found Python version {self.python_version_tag}. Android packaging currently
+requires Python 3.7.""".format(
+                    self=self
+                )
+            )
+
+    def verify_sdk(self):
+        """
+        Install the Android SDK if needed.
+        """
+        tools_path = self.sdk_path / "tools" / "bin"
+        # This method marks some files as executable, so `tools_ok` checks for
+        # that as well. On Windows, all generated files are executable.
+        tools_ok = (
+            tools_path.exists()
+            and all([access(str(tool), X_OK) for tool in tools_path.glob("*")])
+            and (
+                (tools_path / "sdkmanager").exists()
+                or (tools_path / "sdkmanager.exe").exists()
```

Comment:
Your approach makes more sense. The next push to this PR will include this change.

Reply:
Related - it occurred to me while building the template that the gradle script is `gradlew.bat` on windows.

Reply:
True. My current plan is to ignore that until I manually test on Windows.

I think no change will be needed, since Windows will automatically execute `filename.bat` if you tell it to execute `filename` (similar to `filename.exe`).